### ISSUE TITLE
Bot make unhuman moviments

### ISSUE THREAD

**Issue by BoGnY** [PokemonGoF/PokemonGo-Bot#2192]
### Expected Behavior
Walk with a few turns

### Actual Behavior
Walk with tens of turns in 1 meter :S
![bot_walk](https://cloud.githubusercontent.com/assets/11542801/17287508/5baabf52-57d0-11e6-9ee1-c1f78ad719ac.jpg)
This is the max zoom to make the idea of what it does..

### Steps to Reproduce
Launch bot and walk..
Maybe problem should is that my first task is FollowSpiral???
My config file:
```
{
    "auth_service": "google",
    "username": "xxxxxxxxx@gmail.com",
    "password": "xxxxxxxxxxxxxxxxxx",
    "location": "45.494083,12.247316",
    "gmapkey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "tasks": [
      {
        "type": "FollowSpiral"
      },
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "RecycleItems"
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "MoveToFort"
      },
      {
        "type": "EvolveAll",
        "config": {
          "evolve_speed": 20,
          "use_lucky_egg": false
        }
      }
    ],
    "max_steps": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 50
    },
    "websocket_server": false,
    "walk": 11.89,
    "action_wait_min": 1,
    "action_wait_max": 4,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 10,
    "item_filter": {
      "Potion": { "keep" : 75 },
      "Super Potion": { "keep" : 60 },
      "Hyper Potion": { "keep" : 50 },
      "Revive": { "keep" : 60 }
    },
    "evolve_all": "none",
    "evolve_cp_min": 300,
    "evolve_captured": "none",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or" },
      "// Pokemons with example": { "always_catch": true },
      "// Gets filtered with release parameters": {}
    },
    "release": {
      "any": {"release_below_cp": 0, "release_below_iv": 0, "logic": "or"},
      "// Example of always releasing Rattata:": {},
      "// Rattata": {"always_release": true},
      "// Example of keeping 3 stronger (based on CP) Pidgey:": {},
      "// Pidgey": {"keep_best_cp": 3},
      "// Example of keeping 2 stronger (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_iv": 2},
      "// Also, it is working with any": {},
      "// any": {"keep_best_iv": 3},
      "// Example of keeping the 2 strongest (based on CP) and 3 best (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_cp": 2, "keep_best_iv": 3},

      "Abra": {"keep_best_cp": 1},
      "Aerodactyl": {"keep_best_cp": 3},
      "Alakazam": {"keep_best_cp": 2},
      "Arbok": {"keep_best_cp": 2},
      "Arcanine": {"keep_best_cp": 2},
      "Articuno": {"keep_best_cp": 3},
      "Beedrill": {"keep_best_cp": 1},
      "Bellsprout": {"keep_best_cp": 1},
      "Blastoise": {"keep_best_cp": 2},
      "Bulbasaur": {"keep_best_cp": 1},
      "Butterfree": {"keep_best_cp": 1},
      "Caterpie": {"keep_best_cp": 1},
      "Chansey": {"keep_best_cp": 1},
      "Charizard": {"keep_best_cp": 3},
      "Charmander": {"keep_best_cp": 1},
      "Charmeleon": {"keep_best_cp": 2},
      "Clefable": {"keep_best_cp": 2},
      "Clefairy": {"keep_best_cp": 2},
      "Cloyster": {"keep_best_cp": 2},
      "Cubone": {"keep_best_cp": 1},
      "Dewgong": {"keep_best_cp": 1},
      "Diglett": {"keep_best_cp": 1},
      "Ditto": {"keep_best_cp": 2},
      "Dodrio": {"keep_best_cp": 2},
      "Doduo": {"keep_best_cp": 1},
      "Dragonair": {"keep_best_cp": 2},
      "Dragonite": {"keep_best_cp": 3},
      "Drowzee": {"keep_best_cp": 2},
      "Dugtrio": {"keep_best_cp": 2},
      "Eevee": {"keep_best_cp": 3},
      "Ekans": {"keep_best_cp": 1},
      "Electabuzz": {"keep_best_cp": 1},
      "Electrode": {"keep_best_cp": 1},
      "Exeggcute": {"keep_best_cp": 1},
      "Exeggutor": {"keep_best_cp": 2},
      "Farfetch'd": {"keep_best_cp": 1},
      "Fearow": {"keep_best_cp": 2},
      "Flareon": {"keep_best_cp": 3},
      "Gastly": {"keep_best_cp": 1},
      "Gengar": {"keep_best_cp": 2},
      "Geodude": {"keep_best_cp": 1},
      "Gloom": {"keep_best_cp": 1},
      "Golbat": {"keep_best_cp": 1},
      "Goldeen": {"keep_best_cp": 1},
      "Golduck": {"keep_best_cp": 2},
      "Golem": {"keep_best_cp": 1},
      "Graveler": {"keep_best_cp": 2},
      "Grimer": {"keep_best_cp": 2},
      "Growlithe": {"keep_best_cp": 1},
      "Gyarados": {"keep_best_cp": 3},
      "Haunter": {"keep_best_cp": 1},
      "Hitmonchan": {"keep_best_cp": 2},
      "Hitmonlee": {"keep_best_cp": 1},
      "Horsea": {"keep_best_cp": 1},
      "Hypno": {"keep_best_cp": 1},
      "Ivysaur": {"keep_best_cp": 2},
      "Jigglypuff": {"keep_best_cp": 1},
      "Jolteon": {"keep_best_cp": 3},
      "Jynx": {"keep_best_cp": 1},
      "Kabuto": {"keep_best_cp": 1},
      "Kabutops": {"keep_best_cp": 2},
      "Kadabra": {"keep_best_cp": 3},
      "Kakuna": {"keep_best_cp": 1},
      "Kangaskhan": {"keep_best_cp": 2},
      "Kingler": {"keep_best_cp": 2},
      "Koffing": {"keep_best_cp": 2},
      "Krabby": {"keep_best_cp": 1},
      "Lapras": {"keep_best_cp": 2},
      "Lickitung": {"keep_best_cp": 1},
      "Machamp": {"keep_best_cp": 3},
      "Machoke": {"keep_best_cp": 2},
      "Machop": {"keep_best_cp": 1},
      "Magikarp": {"keep_best_cp": 1},
      "Magmar": {"keep_best_cp": 1},
      "Magnemite": {"keep_best_cp": 2},
      "Magneton": {"keep_best_cp": 3},
      "Mankey": {"keep_best_cp": 1},
      "Marowak": {"keep_best_cp": 2},
      "Meowth": {"keep_best_cp": 1},
      "Metapod": {"keep_best_cp": 1},
      "Mew": {"keep_best_cp": 2},
      "Mewtwo": {"keep_best_cp": 3},
      "Moltres": {"keep_best_cp": 2},
      "Mr. Mime": {"keep_best_cp": 1},
      "Muk": {"keep_best_cp": 1},
      "Nidoking": {"keep_best_cp": 3},
      "Nidoqueen": {"keep_best_cp": 3},
      "Nidorina": {"keep_best_cp": 2},
      "Nidorino": {"keep_best_cp": 2},
      "Ninetales": {"keep_best_cp": 1},
      "Oddish": {"keep_best_cp": 1},
      "Omanyte": {"keep_best_cp": 2},
      "Omastar":  {"keep_best_cp": 2},
      "Onix": {"keep_best_cp": 1},
      "Paras": {"keep_best_cp": 1},
      "Parasect": {"keep_best_cp": 1},
      "Persian": {"keep_best_cp": 2},
      "Pidgeot": {"keep_best_cp": 3},
      "Pidgeotto": {"keep_best_cp": 2},
      "Pidgey": {"keep_best_cp": 1},
      "Pikachu": {"keep_best_cp": 1},
      "Pinsir": {"keep_best_cp": 2},
      "Poliwag": {"keep_best_cp": 2},
      "Poliwhirl": {"keep_best_cp": 1},
      "Poliwrath": {"keep_best_cp": 3},
      "Ponyta": {"keep_best_cp": 1},
      "Porygon": {"keep_best_cp": 2},
      "Primeape": {"keep_best_cp": 2},
      "Psyduck": {"keep_best_cp": 1},
      "Raichu": {"keep_best_cp": 2},
      "Rapidash": {"keep_best_cp": 2},
      "Raticate": {"keep_best_cp": 1},
      "Rattata": {"keep_best_cp": 1},
      "Rhydon": {"keep_best_cp": 2},
      "Rhyhorn": {"keep_best_cp": 2},
      "Sandshrew": {"keep_best_cp": 2},
      "Sandslash": {"keep_best_cp": 2},
      "Scyther": {"keep_best_cp": 2},
      "Seadra": {"keep_best_cp": 1},
      "Seaking": {"keep_best_cp": 2},
      "Seel": {"keep_best_cp": 1},
      "Shellder": {"keep_best_cp": 2},
      "Slowbro": {"keep_best_cp": 2},
      "Slowpoke": {"keep_best_cp": 1},
      "Snorlax": {"keep_best_cp": 2},
      "Spearow": {"keep_best_cp": 1},
      "Squirtle": {"keep_best_cp": 1},
      "Starmie": {"keep_best_cp": 2},
      "Staryu": {"keep_best_cp": 1},
      "Tangela": {"keep_best_cp": 1},
      "Tauros": {"keep_best_cp": 1},
      "Tentacool": {"keep_best_cp": 1},
      "Tentacruel": {"keep_best_cp": 2},
      "Vaporeon": {"keep_best_cp": 3},
      "Venomoth": {"keep_best_cp": 2},
      "Venonat": {"keep_best_cp": 1},
      "Venusaur": {"keep_best_cp": 3},
      "Victreebel": {"keep_best_cp": 1},
      "Vileplume": {"keep_best_cp": 1},
      "Voltorb": {"keep_best_cp": 2},
      "Vulpix": {"keep_best_cp": 1},
      "Wartortle": {"keep_best_cp": 2},
      "Weedle": {"keep_best_cp": 1},
      "Weepinbell": {"keep_best_cp": 1},
      "Weezing": {"keep_best_cp": 1},
      "Wigglytuff": {"keep_best_cp": 1},
      "Zapdos": {"keep_best_cp": 3},
      "Zubat": {"keep_best_cp": 1}
    },
    "vips" : {
        "// Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "// any": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },

		"Arcanine": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Articuno": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Blastoise": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Charizard": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Dragonite": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Exeggutor": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Flareon": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Gyarados": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Lapras": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Mew": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Mewtwo": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Moltres": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Muk": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Snorlax": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Vaporeon": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Venusaur": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Weezing": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" },
		"Zapdos": {"catch_above_cp": 1000, "catch_above_iv": 0.5, "logic": "or" }
    }    
}
```

### Other Information
OS: Windows 10 64bit
Git Commit: ff62273291a7e8e3ef13105c1be46a44830be0ba
Python Version: 2.7.12

**Comment by Lunik** [comment]
Same for me. When looking on the logs, the bot try to go to multiple location at the same time.

**Comment by tstumm** [comment]
Move the "FollowSpiral" task to the end. Otherwise it will collide with "MoveToFort"

**Comment by Bashin** [comment]
I had to remove the FollowSpiral task to get the expected behaviour back again

**Comment by tstumm** [comment]
@Bashin That actually not a real solution. You will walk to some forts and if your inventory is full you will just idle.

**Comment by Bashin** [comment]
Bot is running for 2 hours and the path is looking good to me. I know it is not correcting the error but it's a workaround for now. If MoveToFort is enabled and FollowSpiral too the current version of the bot very often tries to reach both destinations even if it results in stepping for and back between fort and spiral direction.
As long as the bot is unable to manage both tasks at the same time without making this mistake  try to remove the spiral part and it could be working again.

**Comment by BoGnY** [comment]
I delete FollowSpiral from my config and bot walk linear perfectly..
But now, I've insert a path config file to make bot walk to specified location, but don't go here..
It make a similar issue, it make tens of turns in a few space, and however not go to specified location..

**Comment by Yanush7** [comment]
BoGnY there was another issue about that, and it is similar case - for now if you are using PATH you need to disable MoveToFort or you will have the bot stuck. If you disable it,  make sure you put enaugh path points in your config because they are the only ones that the bot will visit.

**Comment by Simbe90** [comment]
Same here, I'm on the latest dev, 494bc05, and the bot never reaches the pokestop. I just pretty much get this:

`Moving towards fort xxx, 0.05km left
Arrived at pokestop.
Moving towards fort xxx, 0.04km left
Arrived at pokestop.
Moving towards fort xxx, 0.04km left
Arrived at pokestop.
Moving towards fort xxx, 0.09km left
Moving towards fort xxx, 0.09km left`

It says "arrived at pokestop" but it never actually does. I did a clean install with this dev version and modified the config only by putting my login info in, everything else is on default, including "FollowSpiral" at the end, just after "SpinFort".

I now removed the "FollowSpiral" and everything works as expected.
